gobblin-data-management/src/main/java/gobblin/data/management/retention/action/AccessControlAction.java

```diff
@@ -91,23 +93,58 @@ private void executeOnVersion(DatasetVersion datasetVersion) throws IOException
     // Perform action if it is a FileSystemDatasetVersion
     if (datasetVersion instanceof FileSystemDatasetVersion) {
       FileSystemDatasetVersion fsDatasetVersion = (FileSystemDatasetVersion) datasetVersion;
-      for (Path path : fsDatasetVersion.getPaths()) {
-
-        if (this.fs.exists(path)) {
 
-          // Update permissions if set in config
-          if (this.permission.isPresent()) {
-            FsPermission fsPermission = new FsPermission(this.permission.get());
-            this.fs.setPermission(path, fsPermission);
-            log.debug("Set permissions for {} to {}", path.toString(), fsPermission);
+      // If the version is filestatus aware, use the filestatus to ignore permissions update when the path already has
+      // the desired permissions
+      if (datasetVersion instanceof FileStatusAware) {
+        for (FileStatus fileStatus : ((FileStatusAware)datasetVersion).getFileStatuses()) {
+          if (needsPermissionsUpdate(fileStatus) || needsOwnerUpdate(fileStatus) || needsGroupUpdate(fileStatus)) {
+            updatePermissionsAndOwner(fileStatus.getPath());
           }
+        }
+      } else {
+        for (Path path : fsDatasetVersion.getPaths()) {
+          updatePermissionsAndOwner(path);
+        }
+      }
+    }
+  }
 
-          // Update owner and group if set in config
-          if (this.owner.isPresent() || this.group.isPresent()) {
-            this.fs.setOwner(path, this.owner.orNull(), this.group.orNull());
-            log.debug("Set owner and group for {} to {}:{}", path.toString(), this.owner.orNull(),
-                this.group.orNull());
-          }
+  private boolean needsPermissionsUpdate(FileStatus fileStatus) {
+    return this.permission.isPresent() && !new FsPermission(this.permission.get()).equals(fileStatus.getPermission());
+  }
+
+  private boolean needsOwnerUpdate(FileStatus fileStatus) {
+    return this.owner.isPresent() && this.owner!= null && !StringUtils.equals(owner.get(), fileStatus.getOwner());
+  }
+
+  private boolean needsGroupUpdate(FileStatus fileStatus) {
+    return this.group.isPresent() && this.group!= null && !StringUtils.equals(group.get(), fileStatus.getGroup());
+  }
+
+  private void updatePermissionsAndOwner(Path path) throws IOException {
+    if (this.fs.exists(path)) {
+
+      // Update permissions if set in config
+      if (this.permission.isPresent()) {
+        FsPermission fsPermission = new FsPermission(this.permission.get());
+        if (!this.isSimulateMode) {
+          this.fs.setPermission(path, fsPermission);
```

Comment:
Do we maybe want to catch exception here to make sure change of owner still happens even if `setPermission` failed?